File: Swift/Array/0001_two_sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        // Dictionary [target - nums[i]:index]
        var answer: [Int : Int] = [:]

        for (i, num) in nums.enumerated() {
            // If nums[i] is in answer
            if let index = answer[target - num] {
                return [i, index]
            }

            // If num[i] is not in answer
            answer[num] = i
        }

        return []
    }
}

File: JavaScript/Array/0001_two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    let i = 0;
    let j = 1;
    while (i < nums.length) {
        while (j > i && j < nums.length) {
            if (nums[i] + nums[j] === target) return [i, j];
            j += 1;
        }
        i += 1;
        j = i + 1;
    }
    return [i, j];
};